### PR TITLE
Add locking callback

### PR DESCRIPTION
**Pull request by bingen**

### CODE REVIEW

File: contracts/ILockManager.sol [hunk 1]
```diff
@@ -2,5 +2,6 @@ pragma solidity ^0.4.24;
 
 
 interface ILockManager {
+    function lockCallback(uint256 _amount, uint256 _allowance, bytes _data) external returns (bool);
```
Comment (facuspagnuolo) [inline contracts/ILockManager.sol:5]: What do you think about `receiveLockManager` or similar? I'd add the address of the user as well

File: contracts/ILockManager.sol [hunk 2]
```diff
@@ -2,5 +2,6 @@ pragma solidity ^0.4.24;
 
 
 interface ILockManager {
+    function receiveLockManager(uint256 _amount, uint256 _allowance, bytes _data) external returns (bool);
```
Comment (facuspagnuolo) [inline contracts/ILockManager.sol:5]: Sorry, thinking about this twice I think I would simply call it `receiveLock`

File: package.json [hunk 3]
```diff
@@ -25,6 +26,7 @@
     "solium": "^1.2.3",
     "truffle": "4.1.14",
     "truffle-bytecode-manager": "^1.1.1",
-    "truffle-extract": "^1.2.1"
+    "truffle-extract": "^1.2.1",
+    "web3-utils": "1.2.5"
```
Comment (facuspagnuolo) [inline package.json:30]: Let's create an issue to migrate to Truffle 5

File: contracts/Staking.sol [hunk 4]
```diff
@@ -120,6 +124,10 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         require(_amount <= _unlockedBalanceOf(msg.sender), ERROR_NOT_ENOUGH_BALANCE);
 
         _increaseLockAmountUnsafe(msg.sender, _lockManager, _amount);
+
+        if (_toBytes4(_data) == ILockManager(_lockManager).receiveLockManager.selector) {
+            ILockManager(_lockManager).receiveLockManager(_amount, _allowance, _data);
+        }
```
Comment (facuspagnuolo) [inline contracts/Staking.sol:130]: I think we can probably extract this repeated code to an internal function

File: package.json (hunk 3 above)
Comment (bingen) [reply package.json:30]: https://github.com/aragon/staking/issues/26

**Pull request merged**